### DISCUSSION TITLE
permalinks for messages

### DISCUSSION THREAD

**Discussion by szelga** [the-djmaze/snappymail#874] [Ideas]
is there a feature somewhere to get permalinks for messages and/or folders in SnappyMail? haven't found anything about that in RainLoop and in SnappyMail features list. if not, is there something about that exposed in API, so one could write a plugin implementing that?

**Comment by the-djmaze** [comment]
When you move a message to another folder, the link becomes invalid.
With JMAP each message has an unique ID and then permalinks are possible, but not all mail servers support JMAP.

But if you look carefully to your browser debug window, you will find requests like:
`/?/Json/&q[]=/0/Message/&q[]=/WyJJTkJPWCIsOTc3NywxLCJjN2U1YTE2OWFkMDI1MGE2MzE5YzJhOWU0NGE3ZTViZWYxOWQ0ZTU2Il0`

When you base64 decode the data, you get something like:
`["INBOX",9777,1,"sha1hash"]`

Is that what you are after?

**Reply by szelga** [reply]
sorry for the belated answer. I was after something more permanent, like JMAP. anyway, we're currently tweaking RoundCube in order to have permalinks.

thank you for the answer.
